### ISSUE TITLE
Epic: Update form components to leverage ElementInternals API

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-design-system#8126]
**monday.com sync:** #18048583564

### Description

Lit is looking to [add support for `ElementInternals`](https://github.com/lit/rfcs/pull/45), so we should explore leveraging this API instead of our custom solution. 

cc @eriklharper @benelan 

Based on recent discussion, we'll use this issue to track just replacing the existing `HiddenFormInputSlot` implementation with `ElementInternals` without changing any existing public APIs.  This will keep the refactor label consistent too since no functionality would change as a result.

### Proposed Advantages

Improved form integration and leaner components.

### Which Component

Checkbox, Combobox, Input Date Picker, Input Number, Input Text, Input Time Picker, Input Time Zone, Input, Meter, Radio Button, Rating, Segmented Control, Select, Slider, Switch, Text Area

### Relevant Info

Safari added support for `ElementInternals` at 16.4, so we may have to consider the following if we update form components to use this API:

- [x] waiting until Safari 18 to ensure last 2 **base** versions support this API
- [x] ~consider [a polyfill](https://www.npmjs.com/package/element-internals-polyfill) to handle Safari 16.0 – 16.3 (requires a lot testing)~
- [x]  ~moving forward and document form integration won't work for 16.0 – 16.3 (nonstarter option as it introduces a regression for this version)~

All components implementing `FormComponent`:

* [x] `autocomplete` @jcfranco #14182 
* [x] `checkbox` @eriklharper #14052 
* [x] `combobox` @jcfranco #14056
* [x] `input-date-picker` @jcfranco https://github.com/Esri/calcite-design-system/pull/14072
* [x] `input-number` @jcfranco https://github.com/Esri/calcite-design-system/pull/14072
* [x] `input-text` @eriklharper #13985 
* [x] `input-time-picker` @jcfranco https://github.com/Esri/calcite-design-system/pull/14072
* [x] `input-time-zone` @jcfranco https://github.com/Esri/calcite-design-system/pull/14135
* [x] `input` @jcfranco https://github.com/Esri/calcite-design-system/pull/14073
* [x] ~`meter` @jcfranco~ [no longer applicable](https://github.com/Esri/calcite-design-system/pull/14071)
* [x] `radio-button` @eriklharper https://github.com/Esri/calcite-design-system/pull/14142
* [x] `rating`  @jcfranco https://github.com/Esri/calcite-design-system/pull/14069
* [x] `segmented-control` @eriklharper https://github.com/Esri/calcite-design-system/pull/14125
* [x] `select` @eriklharper #14066 
* [x] `slider` @jcfranco https://github.com/Esri/calcite-design-system/pull/14069
* [x] `switch`  @jcfranco https://github.com/Esri/calcite-design-system/pull/14069
* [x] `text-area` @jcfranco https://github.com/Esri/calcite-design-system/pull/14074

Components that can be used to trigger form submit and reset events:

* [x] `action` @eriklharper #14015 
* [x] `button` @eriklharper #14015 

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Related issues

* https://github.com/Esri/calcite-design-system/issues/5026
* https://github.com/Esri/calcite-design-system/issues/4506
* https://github.com/Esri/calcite-design-system/issues/2838
* https://github.com/Esri/calcite-design-system/issues/4078
* https://github.com/Esri/calcite-design-system/issues/7249
* https://github.com/Esri/calcite-design-system/issues/6751
* https://github.com/Esri/calcite-design-system/issues/4681
* https://github.com/Esri/calcite-design-system/issues/7413
* https://github.com/Esri/calcite-design-system/issues/9319 `high` `support`
* https://github.com/Esri/calcite-design-system/issues/11148

**Comment by geospatialem** [comment]
With the transition to Lit from Stencil, Calcite is ensuring stability before proceeding with `ElementInternals` support, targeting the above in the next upcoming milestone (December 2024).

**Comment by jcfranco** [comment]
This is blocked by https://github.com/lit/rfcs/pull/45.

**Comment by eriklharper** [comment]
PR https://devtopia.esri.com/WebGIS/arcgis-web-components/pull/10359 has been merged, this is ready for development!

**Comment by DintaMel** [comment]
@eriklharper do you have any guidance on how we might verify this?

**Comment by eriklharper** [comment]
> [@eriklharper](https://github.com/eriklharper) do you have any guidance on how we might verify this?

Good question!  I'll sync up with you directly to develop a good strategy to verify this, since there's a quite a lot of components and it will be worth it to document in more detail the types of things we should be testing with this update.

**Comment by eriklharper** [comment]
Here's a comprehensive demo codepen you can use to verify: https://codepen.io/editor/eriklharper/pen/019dfee4-f2b2-7b90-a330-275d7be77b5d

Basic instruction is to test the 2 types of submit buttons (one calcite, one action), the 2 types of reset buttons (one calcite, one action) and make sure the form stops submit for each invalid component.  They're all marked as `required` so you'll have to fill in each one with the exception of Time Zone and Slider which already have a default value.  You can test required validation for Time Zone by clearing the value before submit.

Open up the console in codepen UI to see that on invalid submit, it logs out each invalid component, and it will log out all the form data with the key corresponding to the `name` set on each element and the resulting value its set to as a JSON string.

**Comment by DintaMel** [comment]
@eriklharper Thanks for the codepen! It's super helpful. I'm still testing, but wanted to ask about this use case.

For Radio Button Group, if it's `required` I see a console error if I make one of the child Radio Buttons `disabled`. [Codepen](https://codepen.io/editor/DintaMel-the-lessful/pen/019e046d-70e1-777e-b407-595861172aee) Can you take a look?

<img width="1357" height="290" alt="Image" src="https://github.com/user-attachments/assets/96063c16-510a-4111-812f-26b84f692e2e" />

**Comment by DintaMel** [comment]
@eriklharper I noticed a couple of differences in Slider that I wanted to run by you. See[ codepen with examples](https://codepen.io/editor/DintaMel-the-lessful/pen/019e0820-56c6-76ff-940c-f6c6444c119f).

- Zero value: In 5.0, zero is not a valid submission. In `@next` it is valid. I think that zero should be valid, so I think this is just an unintentional bug fix.

- Multiple handles: If there are two handles both on the same value, in 5.0 we only submitted a singular value. In `@next` both handles are submitted. Note: I edited the initial codepen JS so that it got all the data for the key, otherwise it only showed the most recently changed handle value.

**Comment by DintaMel** [comment]
I verified reset, submit, disabled, and required functionality with `5.1.0-next.47`. Everything besides the above comments looked good to me.

@DitwanP  is testing the range based verifications.

**Comment by eriklharper** [comment]
> For Radio Button Group, if it's `required` I see a console error if I make one of the child Radio Buttons `disabled`. [Codepen](https://codepen.io/editor/DintaMel-the-lessful/pen/019e046d-70e1-777e-b407-595861172aee) Can you take a look?

#14400

**Comment by DitwanP** [comment]
Verified range based validation on `5.1.0-next.47` Everything is behaving the same as on `5.0`.

**Comment by eriklharper** [comment]
> Multiple handles: If there are two handles both on the same value, in 5.0 we only submitted a singular value. In @next both handles are submitted. Note: I edited the initial codepen JS so that it got all the data for the key, otherwise it only showed the most recently changed handle value.

@jcfranco would you consider this a bug or an improvement?

**Comment by jcfranco** [comment]
Seems like a fix to me. In a range scenario, I would always expect min and max, regardless of the value.

**Comment by DintaMel** [comment]
@eriklharper the `setValidity` issue looks good now. I noticed another issue while testing. The Radio Button Group `validationMessage` is missing in `5.1.0-next.54`. See [codepen](https://codepen.io/editor/DintaMel-the-lessful/pen/019e046d-70e1-777e-b407-595861172aee).

5.0
<img width="429" height="64" alt="Image" src="https://github.com/user-attachments/assets/f4e9f4c7-1389-43f4-8b54-935ffa4c416d" />

5.1.0-next.54

<img width="406" height="44" alt="Image" src="https://github.com/user-attachments/assets/64836733-d579-4d3a-885b-2710465287bc" />

**Comment by isaacbraun** [comment]
Verified on `5.1.0-next.62` that validity, submit, reset, and disabled functionality is working as intended via [CodePen](https://codepen.io/editor/isaacbraun/pen/019e6c40-2634-767f-8ab3-6240e0de9e4a) and Mel's testing [(comment)](https://github.com/Esri/calcite-design-system/issues/8126#issuecomment-4408502584). 🎉
